### DISCUSSION TITLE
laravel collection contains not working properly

### DISCUSSION THREAD

**Discussion by joisarjignesh** [laravel/framework#33168] [General]
- Laravel Version: 7.0
- PHP Version: 7.4.6
- Database Driver & Version: sqlite

### Description: laravel collection contains method not working properly

for example :
```
$collection = collect([
    ['color' => 'blue', 4, 2],
    ['x', 'y', 'color' => 'red', 'shape' => 'circle', 300],
    ['make' => 'Acura', 'type' => 'NSX'],
]);

dd($collection->contains('color', 'blue'));   //return true
dd($collection->collapse()->contains('color', 'blue')); //return false
```

without collapse() contains work properly after collapse() and then use contains() it not working properly
it will return result as false but 'color' key  with value 'blue' is there .

**Comment by driesvints** [comment]
Moving to discussions.

**Comment by devcircus** [comment]
Just used tinker to run your code. When you collapse that collection, 'color' => 'blue' is overwritten by 'color' => 'red', so it is correct:
```php
$col = collect([
    ['color' => 'blue', 4,2],
    ['x', 'y', 'color' => 'red', 'shape' => 'circle', 300],
    ['make' => 'Acura', 'type' => 'NSX']
]);
$col->contains('color', 'blue'); //true

$collapsed = $col->collapse();
dump($collapsed);
Illuminate\Support\Collection {#3075
     all: [
       "color" => "red",
       0 => 4,
       1 => 2,
       2 => "x",
       3 => "y",
       "shape" => "circle",
       4 => 300,
       "make" => "Acura",
       "type" => "NSX",
     ],
   }

$collapsed->contains('color', 'blue'); //false
```

**Comment by joisarjignesh** [comment]
in above code, if you remove **'color =>  red'**  entry and use collapse still it returns false.
```

$col = collect([
    ['color' => 'blue', 4,2],
    ['x', 'y', 'shape' => 'circle', 300],
    ['make' => 'Acura', 'type' => 'NSX']
]);


$col->collapse()->contains('color', 'blue'); //returning false
```

**Comment by devcircus** [comment]
Yes it looks like "contains" expects a collection of object-like elements, ie:
```php
$col = collect([
    ['name' => 'Bob', 'age' => 30, 'color' => 'blue'],
    ['name' => 'Anne', 'age' => 28, 'color' => 'red'],
    ['name' => 'Sally', 'age' => 29, 'color' => 'purple'],
]);

$col->contains('name', 'Anne'); //true
$col->contains('age', 41); //false
```
With a simple flat array, ```$myArr['color'] === 'blue';``` will suffice.

If you dig into the source for ```contains```, it checks each inner element to see if it contains the arguments, instead of checking the outer collection. So the first time through, it checks the value of the first element, which is just "blue". "blue" doesn't contain a "color" attribute, so it returns false.
